# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieNotification.cpp
# Commit: 5aa3b2f1b5ec
# Full Hash: 5aa3b2f1b5ec8a63fa04f80cf4aed4aac2c6ce22
# Author: Paul Zuehlcke <pbz@mozilla.com>
# Date: 2023-09-07 21:34:02
# Regressor Bug: 1839918
# File Overlap Count: 6
# Description:
#   Bug 1839918 - Refactor "cookie-changed" notifications to use nsICookieNotification. r=extension-reviewers,cookie-reviewers,sessionstore-reviewers,bvandersloot,edgul,robwu
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D185300
# ==============================================================================

diff -r fecbf97cc8db -r 5aa3b2f1b5ec netwerk/cookie/CookieNotification.cpp
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/netwerk/cookie/CookieNotification.cpp	Thu Sep 07 08:25:54 2023 +0000
@@ -0,0 +1,66 @@
+/* -*- Mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "CookieNotification.h"
+#include "mozilla/dom/BrowsingContext.h"
+#include "nsICookieNotification.h"
+
+namespace mozilla::net {
+
+NS_IMETHODIMP
+CookieNotification::GetAction(nsICookieNotification::Action* aResult) {
+  *aResult = mAction;
+  return NS_OK;
+}
+
+NS_IMETHODIMP
+CookieNotification::GetCookie(nsICookie** aResult) {
+  NS_ENSURE_ARG_POINTER(aResult);
+
+  *aResult = mCookie;
+  NS_IF_ADDREF(*aResult);
+
+  return NS_OK;
+}
+
+NS_IMETHODIMP CookieNotification::GetBaseDomain(nsACString& aBaseDomain) {
+  aBaseDomain = mBaseDomain;
+
+  return NS_OK;
+}
+
+NS_IMETHODIMP
+CookieNotification::GetBatchDeletedCookies(nsIArray** aResult) {
+  NS_ENSURE_ARG_POINTER(aResult);
+  NS_ENSURE_TRUE(mAction == nsICookieNotification::COOKIES_BATCH_DELETED,
+                 NS_ERROR_NOT_AVAILABLE);
+
+  *aResult = mBatchDeletedCookies;
+  NS_IF_ADDREF(*aResult);
+
+  return NS_OK;
+}
+
+NS_IMETHODIMP
+CookieNotification::GetBrowsingContextId(uint64_t* aResult) {
+  *aResult = mBrowsingContextId;
+  return NS_OK;
+}
+
+NS_IMETHODIMP
+CookieNotification::GetBrowsingContext(dom::BrowsingContext** aResult) {
+  *aResult = dom::BrowsingContext::Get(mBrowsingContextId).take();
+  return NS_OK;
+}
+
+NS_IMETHODIMP
+CookieNotification::GetIsThirdPartyCookie(bool* aResult) {
+  *aResult = mIsThirdPartyCookie;
+  return NS_OK;
+}
+
+NS_IMPL_ISUPPORTS(CookieNotification, nsICookieNotification)
+
+}  // namespace mozilla::net